File: csharp-runtime/RpcException.cs
namespace RosettaRpc;

/// <summary>
/// Thrown by <see cref="RpcClient.Call{TReq,TResp}"/> when a call fails —
/// either at the transport/framework level (unknown service/method, decode
/// failure) or because the service implementation itself failed. Callers
/// distinguish these with <see cref="Code"/>.
/// </summary>
public sealed class RpcException : Exception
{
    public RpcErrorCode Code { get; }

    public RpcException(RpcErrorCode code, string message) : base(message)
    {
        Code = code;
    }
}

File: kotlin-runtime/src/main/kotlin/rosettarpc/RpcException.kt
package rosettarpc

/**
 * Thrown by [RpcClient.call] when a call fails — either at the transport/framework level (unknown
 * service/method, decode failure) or because the service implementation itself failed. Callers
 * distinguish these with [code].
 */
class RpcException(val code: RpcErrorCode, message: String) : RuntimeException(message)
